scripts/windows/setup/IotEdgeSecurityDaemon.ps1

```diff
@@ -641,52 +658,58 @@ function Remove-SecurityDaemonResources {
 
                 # 2. Reset their ACLs so that they inherit from their container
-                Invoke-Native "icacls ""$MobyDataRootDirectory"" /reset /t /l /q /c"
+                Invoke-Native "icacls ""$root"" /reset /t /l /q /c"
 
                 # 3. Use cmd's `rd` rather than `Remove-Item` since the latter gets tripped up by reparse points, etc.
                 #    Prepend the path with `\\?\` since the layer directories have long names, so the paths usually exceed 260 characters,
                 #    and IoT Core's filesystem doesn't seem to automatically use (or even have) short names
-                Invoke-Native "rd /s /q ""\\?\$MobyDataRootDirectory"""
+                Invoke-Native "rd /s /q ""\\?\$root"""
 
-                Write-Verbose "Deleted Moby data root directory '$MobyDataRootDirectory'"
+                Write-Verbose "Deleted Moby data root directory '$root'"
             }
             catch {
                 Write-Verbose "$_"
-                Write-HostRed ("Could not delete Moby data root directory '$MobyDataRootDirectory'. Please reboot " +
+                Write-HostRed ("Could not delete Moby data root directory '$root'. Please reboot " +
                     'your device and run `Uninstall-SecurityDaemon` again with `-Force`.')
                 $success = $false
             }
         }
     }
     else {
-        Write-Host "Not deleting Moby data root directory '$MobyDataRootDirectory' since -DeleteMobyDataRoot was not specified."
+        Write-Host "Not deleting Moby data root directory since -DeleteMobyDataRoot was not specified."
     }
+    
+    $existingMobyInstallations = $MobyInstallDirectory, $MobyStaticInstallDirectory | ? {test-path $_}
 
-    Remove-Item -Recurse $MobyInstallDirectory -ErrorAction SilentlyContinue -ErrorVariable cmdErr
-    if ($?) {
-        Write-Verbose "Deleted install directory '$MobyInstallDirectory'"
-    }
-    elseif ($cmdErr.FullyQualifiedErrorId -ne 'PathNotFound,Microsoft.PowerShell.Commands.RemoveItemCommand') {
-        Write-Verbose "$cmdErr"
-        Write-HostRed ("Could not delete install directory '$MobyInstallDirectory'. Please reboot " +
-            'your device and run `Uninstall-SecurityDaemon` again with `-Force`.')
-        $success = $false
-    }
-    else {
-        Write-Verbose "$cmdErr"
+    foreach ($install in $existingMobyInstallations) {
+        Remove-Item -Recurse $install -ErrorAction SilentlyContinue -ErrorVariable cmdErr
+        if ($?) {
+            Write-Verbose "Deleted install directory '$install'"
+        }
+        elseif ($cmdErr.FullyQualifiedErrorId -ne 'PathNotFound,Microsoft.PowerShell.Commands.RemoveItemCommand') {
+            Write-Verbose "$cmdErr"
+            Write-HostRed ("Could not delete install directory '$install'. Please reboot " +
+                'your device and run `Uninstall-SecurityDaemon` again with `-Force`.')
+            $success = $false
```

Comment:
Since this is now inside a loop rather than an if block, should we break out early after setting `$success = $false`? Otherwise `$success` would need to be a boolean AND of the result of each loop iteration...

Reply:
Yeah, I think breaking the loop makes the most sense. If we're asking the user to re-run, then there's no harm in stopping here. I'll adjust it.

Reply:
Actually, does it matter which iteration of the loop failed? Failure is expressed outside the scope of the loop and a failure of either iteration would lead us to the same place (reporting failure for the function). We don't set success to true within the scope of the loop so our worst case scenario is that we then attempt to delete two directories and report failure of both (or silently succeed at one, which doesn't need to be repeated on a second run).

Reply:
I think I prefer it to be a loop, so that it attempts to delete the second path even if the first one failed.

Reply:
You're right...I was worried about a `$false` iteration being masked by a subsequent `$true` iteration, but the loop never sets `$success = $true` so we're ok.